File: hashtable/prob1742.py
class Solution:
    def countBalls(self, lowLimit: int, highLimit: int) -> int:
        
        
        def sumDigits(n):
            
            s = 0
            while n > 0:
                n, r = divmod(n, 10)
                s += r
                
            return s
                
        d = {}
        for i in range(lowLimit, highLimit+1):
            s = sumDigits(i)
            d[s] = d.get(s, 0) + 1
            
        return max(d.values())

File: math/prob1742.kt
// 1742. Maximum Number of Balls in a Box
class Solution {
    fun countBalls(lowLimit: Int, highLimit: Int): Int {

        var counter = mutableMapOf<Int, Int>()
        var tmp: Int
        for (n in lowLimit..highLimit) {
            tmp = helper(n)
            counter.put(tmp, counter.getOrDefault(tmp, 0) + 1)
        }

        return counter.values.max()!!


    }

    fun helper(n: Int): Int {
        var ans = 0
        var m = n
        while (m > 0) {
            ans += m % 10
            m /= 10
        }
        return ans
    }
}
